Source/Task/CriticalThread.cpp

```diff
@@ -76,23 +41,12 @@ STDAPI SetTimeCriticalThread(_In_ bool isTimeCriticalThread)
 /// </summary>
 STDAPI VerifyNotTimeCriticalThread()
 {
-    BOOL pending;
-    if (!InitOnceBeginInitialize(&s_tlsInit, INIT_ONCE_CHECK_ONLY, &pending, nullptr) || pending)
+    if ((tls_threadState & CRITICAL_TRUE) == 0)
     {
-        // This thread is either still initializing or was never initialized.  It
-        // can't be time critical.
         return S_OK;
     }
 
-    ULONG_PTR timeCritical = reinterpret_cast<ULONG_PTR>(TlsGetValue(s_tlsSlot));
-
-    if ((timeCritical & CRITICAL_TRUE) == 0)
-    {
-        DWORD error = GetLastError();
-        return HRESULT_FROM_WIN32(error);
-    }
-
-    RETURN_HR(E_TIME_CRITICAL_THREAD);
+    RETURN_HR(E_HC_TIME_CRITICAL_THREAD);
```

Comment:
I would prefer to define all these as general "windows E_" errors like E_INVALIDARG.  That way we can share values / constant names with Windows (we will use the same names).-